secretstores/local/keyring/keyring_test.go

```diff
@@ -0,0 +1,382 @@
+/*
```

Comment:
Great that you added tests! But, would you consider moving these to be certification tests?

I am concerned with two things:

1. On local machines, if users don't have the dependencies installed (gnupg etc) tests may fail when they run `make test`. We want to make sure that contributors have the best experience even when developing locally, and we should avoid adding dependencies for running unit tests
2. On the CI, installing the dependencies will slow down the test runs for a bit.

These tests are fairly comprehensive already and I think they would be great as certification tests